ciphers/xor_cipher.cpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file xor_cipher.cpp
+ * @brief Implementation of [XOR cipher](https://en.wikipedia.org/wiki/XOR_cipher) algorithm.
+ *
+ * @details
+ * In cryptography, the simple XOR cipher is a type of additive cipher, an encryption 
+ * algorithm that operates according to the principles: 
+ *
+ * \f[A {\oplus} 0 = A\f]
```

Comment:
can be better presented as an eqnarray
```latex
\f{eqnarray*}{
A {\oplus} 0 &=& A \\
A {\oplus} A &=& 0 \\
A {\oplus} B) {\oplus} C &=& A {\oplus} (B {\oplus} C) \\
(B {\oplus} A) {\oplus} B &=& B {\oplus} 0 = B
\f}
```
 or as bullet points
  * \f[A {\oplus} 0 = A\f]
  * \f[A {\oplus} A = 0\f]
  * \f[ (A {\oplus} B) {\oplus} C = A {\oplus} (B {\oplus} C)\f]
  * \f[ (B {\oplus} A) {\oplus} B = B {\oplus} 0 = B \f]